File: algorithms/median.c
#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>

int list[5];
int median;
int median_index;



// NOTE: learn for loop
// NOTE: sizeof() function
void print_list() {
	int i;
	for (i = 0; i < sizeof(list); i ++) {
		printf("%d ", list[i]);
	}
	printf("\n");
}



void get_list() {
	printf("Enter 5 integers:\n");
	int num_inputs = scanf("%d %d %d %d %d", list, &list[1], &list[2], &list[3], &list[4]); 
}



// NOTE: start here after the code puzzle
//	 - learn for loop, nested for loop
//	 - using iterators
//	 - brainstorm sorting algorithm and implement
int my_sort_list() {
	print_list();

	int i, j, x, y;
	for (i = 1; i < sizeof(list); i ++) {
		x = list[i];
		for (j = i - 1; j >= 0; j--) {
			y = list[j];
			if (x < y) {
				list[j] = x;
				list[j+1] = y;
			}
			print_list();
		}
	}

	return 0;
}


// NOTE: - compute median
//	 - learn data sizes
void get_median() {
	median_index = (sizeof(list)/sizeof(int))/2;
	median = list[median_index];
}


int main() {
	get_list();
	printf("Successfully read integer list: ");
	print_list();

	sort_list();
	printf("Successfully sorted list: "),
	print_list();

	get_median();
	printf("\n Median: %d\n", median);
	
	return 0;
}

File: algorithms/median.py
int_list = [0] * 5
median = 0
median_index = 0

def get_list():
	# NOTE: need to declare global sometimes
	global int_list

	print("Enter 5 integers: ")
	str_list = input().split()
	# NOTE: learning for loop syntax
	#	--> learning range function
	for i in range(0, 5):
		int_list[i] = int(str_list[i])

def print_list():
	global int_list
	# NOTE: python print function supports list printing
	print(int_list)


def sort_list():
	global int_list

	# NOTE: basic sorting algorithm	
	print("Sorting input list ...")
	for i in range(1, 5):
		x = int_list[i]
		for j in range(i - 1, -1, -1):
			y = int_list[j]
			if (x < y):
				int_list[j] = x
				int_list[j + 1] = y
			print_list()
	

def get_median():
	global median
	global median_index

	median_index = int(len(int_list)/2)
	median = int_list[median_index]


def main():
	get_list()
	print("Input list = ")
	print_list()
	sort_list()
	print("Sorted list = ")
	print_list()
	get_median()
	print("Median = ", median)


main()



